### ISSUE TITLE
Extension IPC

### ISSUE THREAD

**Issue by nls1729** [micheleg/dash-to-dock#291]
I have a user who reported a bug on my extension activities-config@nls1729.  Each time he booted the setting Hide Panel Round Corners was set from ON to OFF.  We found the Dash to Dock was hiding and showing the corners on each enable and when the position setting changed.  I made a workaround which I really don't like.  It involves catching the "show" signal and re-hiding the corners.  I looked at your code and thought it may be better to fix the problem in Dash to Dock in a 100% reliable way.  I have toyed with the idea of a means for extensions to cooperate.  I adapted some of my experimental code to allow the Dash to Dock to test the setting in my extension and only do the "hide" and "show" if the setting indicated the OFF state.  The code is very simple and appears to work quite well.  It is coded to be bullet proof but could be re-done to execute more efficently.  From my testing the code does not need to be efficent since it is not executed very often in Dash to Dock.  To get to the point I would like for you to look at the code and consider including it in Dash to Dock.  Let me know what you think.

**Comment by micheleg** [comment]
Hi,

I'm definitely for allowing the best interoperability between extensions, as long as this does not makes the extensions more complicated to maintain. Unfortunately there always be cases with extension patching the same components.

For the specific issue there might be a simpler way to avoid the problem. I'm tweaking the round corners in principle for the panel mode only, but those settings get reset on each extension reload including position change). I have to think about it.

I'm curious to have a look at your code, please send it over. 

Michele

**Comment by nls1729** [comment]
Hello,
Thanks for your quick response.

See the attached patch.  The Activities Configurator gives the user the option to hide the rounded corners.  If the corners are hidden Dash to Dock need not fiddle with them.  The classic solution is IPC. This patch provides a crude semaphore.

Norman


[patch.txt](https://github.com/micheleg/dash-to-dock/files/165928/patch.txt)

**Comment by micheleg** [comment]
I had a quick look at your patch. You are basically proposing that an extension check for the existence of another one before tweaking some aspects of the shell. I'm ok to include such mechanism in dash to dock, i'm wondering which is the best way to achieve this. Did you do some testing on the way you retrieve the existence of other extensions? Are you taking into account the case in which the extension is installed but not running?

**Comment by nls1729** [comment]
After testing with the Dash to Dock and Activities Configurator I conclude the only sure way to avoid the conflict between your extension and my extension is to handle the problem within my code for me and, if you choose, you to handle it in your code.  The hide and show problem exists in both our extensions.  I found cases where I show the corners when you have them hidden as you show the corners when I have did hidden.  The only sensible solution is handle it with a signal within the respective extensions.  I connect to the 'show' signal of one corner and when caught, wait one second and then re-hide.  I have some code to detect deadly embraces if someone ever connects to the hide signal (for some unknown reason) and re-shows.  The Activities Configurator only shows when it is disabled and the preference is ON, or the preference is manually changed from ON to OFF.  If you have heard no complaints it is likely the conflict is rarely if ever noticed by your users.
